tests/drivers/rtc/rtc_api/testcase.yaml

```diff
@@ -6,3 +6,9 @@ tests:
     tags: drivers rtc api
     filter: dt_alias_exists("rtc")
     depends_on: rtc
+  drivers.rtc.rtc_api.pcf8523:
```

Comment:
The `build_all` folders are especially useful when no board _nor_ shield has the device being build tested.

Reply:
The purpose of the `rtc_api` is to test the RTC API. Building the driver serves a different purpose, with different requirements. I really prefer separating the two, into a new project named `rtc_build_only` alongside the `rtc_api` project.

This will make the build time shorter for the build only test cases, since they don't have to include the test suite or its dependencies, and indicate the purpose of each project by the project (folder) name :)

Reply:
Fair point. I have changed it to be a build-only test under `tests/drivers/build_all/rtc/` (as this is where all the other `build_only` projects live).

Reply:
Awesome :)